src/Psy/Shell.php

```diff
@@ -203,6 +218,34 @@ protected function getDefaultCommands()
     }
 
     /**
+     * @return array
+     */
+    protected function getTabCompletionMatchers()
+    {
+        $this->addTabCompletionMatchers(
```

Comment:
maybe add an `if (empty($this->tabCompletionMatchers))` guard here? I know it's protected, so we don't need to worry about random users calling it a second time, but that'll help guard against future changes doing something dumb and re-initializing these :)